src/data-sources/intermine/api/get-data-sets.js

```diff
@@ -0,0 +1,99 @@
+// get DataSets for a BioEntity, GeneticMap, LinkageGroup, Location, OntologyAnnotation, Ontology, OntologyTerm, Pathway, Phylotree, SyntenyBlock
+async function getDataSets({bioEntity, geneticMap, linkageGroup, location, ontologyAnnotation, ontology, ontologyTerm, pathway, phylotree, syntenyBlock, start=0, size=10}={}) {
```

Comment:
I recommend breaking this into multiple functions - one for each data set type - since requesting multiple types will only return the first matched by your if-else blocks.

Reply:
This is actually a practice I've used throughout - the assumption being that you're only requesting for one type, because I don't see a situation where you'd resolve dependents of an object with multiple types of _that_ object. If you can provide a use case where getDataSets would be called with multiple parameters, I'm all for splitting it out, but I really did _not_ want to create a nearly identical function for each type that would be called. That  use case appears nowhere in the current graphql-server and it's pretty highly functional.

Reply:
By the way, same topic again, DataSet has a bioEntities reverse reference, so you can get the DataSets for any BioEntity via that. But objects that don't extend BioEntity usually only have a forward dataSets collection, in which case you have to query that object for the DataSets, which is why this is so clunky, with the extra query attributes having to be defined as well. I think the only option would be to massively update the mine datamodel to load DataSet with tons of reverse references to those objects, which is even worse, I think, since it's not motivated by any mine purposes.